mirrord/protocol/src/tcp.rs

```diff
@@ -830,3 +925,23 @@ impl<E> TryFrom<InternalHttpResponse<Vec<u8>>> for Response<BoxBody<Bytes, E>> {
         ))
     }
 }
+
+impl<E> TryFrom<InternalHttpResponse<HttpResponseStreamed>> for Response<BoxBody<Bytes, E>> {
+    type Error = http::Error;
+
+    fn try_from(value: InternalHttpResponse<HttpResponseStreamed>) -> Result<Self, Self::Error> {
+        let InternalHttpResponse {
+            status,
+            version,
+            headers,
+            body,
+        } = value;
+
+        let mut builder = Response::builder().status(status).version(version);
+        if let Some(h) = builder.headers_mut() {
+            *h = headers;
+        }
+
+        builder.body(BoxBody::new(body.map_err(|_| unreachable!())))
```

Comment:
`HttpResponseStreamed` can actually yield a `hyper::Error`, so it's not really `unreachable!` :V